src/stacks-hierarchy/Root_PurchaseOverviewScreen/use-offer-state.ts

```diff
@@ -93,11 +111,21 @@ const getOfferReducer =
           userProfilesWithCountAndOffer: [],
         };
       case 'SET_OFFER':
+        const totalPrice = calculateTotalPrice(
+          userProfilesWithCounts,
+          action.offers,
+        );
+        const originalPrice = calculateOriginalPrice(
+          userProfilesWithCounts,
+          action.offers,
+        );
+
         return {
           ...prevState,
           offerSearchTime: Date.now(),
           isSearchingOffer: false,
           validDurationSeconds: getValidDurationSeconds(action.offers?.[0]),
+          originalPrice: originalPrice !== totalPrice ? originalPrice : undefined,
```

Comment:
checks if `original price` is equal to `total price`, if it is equal, then there are no discounts, thus we leave it undefined. Otherwise, if the `original price` is different to `total price`, then show the original price.

Reply:
@rosvik the response from back-end is `original_amount`, therefore I named it `originalPrice` to keep the naming consistent.

As for the view logic to show the original price, that does make sense, thanks!

Reply:
updated